File: src/main/java/com/orange/dojo/elephantcarpaccio/terminal/Terminal.java
package com.orange.dojo.elephantcarpaccio.terminal;

import java.io.PrintWriter;
import java.util.Scanner;

public class Terminal {
  private PrintWriter output;
  private NumberReader numberReader;
  private TextReader textReader;

  public Terminal() {
    this(new Scanner(System.in), new PrintWriter(System.out, true));
  }

  Terminal(Scanner input, PrintWriter output) {
    // To change the local
    //input.useLocale(Locale.US);
    this.output = output;
    this.numberReader = new NumberReader(input);
    this.textReader = new TextReader(input);
  }

  public int whatIsYourAge() {
    display("What is your age? :");
    return numberReader.readInteger();
  }

  public String whatIsYourName() {
    display("What is your name? :");
    return textReader.readWord();
  }

  public void display(String message) {
    output.println(message);
  }
}

File: kotlin/src/main/kotlin/com/orange/dojo/elephantcarpaccio/terminal/Terminal.kt
package com.orange.dojo.elephantcarpaccio.terminal

import java.io.PrintWriter
import java.util.*

class Terminal internal constructor(input: Scanner, private val output: PrintWriter) {
    private val numberReader: NumberReader
    private val textReader: TextReader

    constructor() : this(Scanner(System.`in`), PrintWriter(System.out, true))

    init {
        // To change the local
        //input.useLocale(Locale.US);
        this.numberReader = NumberReader(input)
        this.textReader = TextReader(input)
    }

    fun whatIsYourAge(): Int {
        display("What is your age? :")
        return numberReader.readInteger()
    }

    fun whatIsYourName(): String {
        display("What is your name? :")
        return textReader.readWord()
    }

    fun display(message: String) {
        output.println(message)
    }
}
